src/redis-cli.c

```diff
@@ -7768,24 +7777,29 @@ unsigned long long sendSync(redisContext *c, char *out_eof, int send_sync) {
     return strtoull(buf+1,NULL,10);
 }
 
-static void slaveMode(int is_psync) {
+static void slaveMode(int send_sync) {
     static char eofmark[RDB_EOF_MARK_SIZE];
     static char lastbytes[RDB_EOF_MARK_SIZE];
     static int usemark = 0;
-    unsigned long long payload = sendSync(context,eofmark,!is_psync);
+    static int out_full_mode;
+    unsigned long long payload = sendSync(context,eofmark,&out_full_mode,send_sync);
     char buf[1024];
     int original_output = config.output;
-    char *command = is_psync ? "PSYNC" : "SYNC";
+    char *command = out_full_mode ? "SYNC" : "PSYNC";
 
-    if (payload == 0 && !is_psync) {
+    if (out_full_mode == 1 && payload == 0) {
+        /* SYNC with EOF marker. */
         payload = ULLONG_MAX;
         memset(lastbytes,0,RDB_EOF_MARK_SIZE);
         usemark = 1;
         fprintf(stderr,"%s with master, discarding "
                        "bytes of bulk transfer until EOF marker...\n", command);
-    } else {
+    } else if (out_full_mode == 1 && payload != 0) {
+        /* SYNC without EOF marker or PSYNC +FULLRESYNC. */
         fprintf(stderr,"%s with master, discarding %llu "
                        "bytes of bulk transfer...\n", command, payload);
+    } else if (out_full_mode == 0 && payload == 0) {
+        /* PSYNC +CONTINUE (no RDB payload). */
```

Comment:
if we have a printf in both other cases, let's print something in this one too.

Reply:
got it

Reply:
got it